OSDOCS:13467: Creating new tutorial book and updating CLI tutorial (#90215)
openshift/openshift-docs: tutorials/dev-app-cli.adoc

:_mod-docs-content-type: ASSEMBLY
[id="dev-app-cli"]
= Tutorial: Deploying an application by using the CLI
include::_attributes/common-attributes.adoc[]
:context: dev-app-cli

toc::[]

This tutorial walks you through deploying services to stand up an application called `national-parks-app` on {product-title} that displays a map of national parks across the world. You will use the {oc-first} to complete this tutorial.

To complete this tutorial, you will perform the following steps:

. xref:../tutorials/dev-app-cli.adoc#getting-started-cli-creating-new-project_dev-app-cli[Create a new project for the application].
+
This step allows your application to be isolated from other cluster user's workloads.

. xref:../tutorials/dev-app-cli.adoc#getting-started-cli-granting-permissions_dev-app-cli[Grant view permissions].
+

> This step grants view permissions to help discover services and other resources running within the project.

Reviewer: Maybe something like:

This step grants view permissions to properly interact with the OpenShift AI, thus helping to discover services and other resources running within the project.

:_mod-docs-content-type: ASSEMBLY
[id="dev-app-cli"]
= Tutorial: Deploying an application by using the CLI
include::_attributes/common-attributes.adoc[]
:context: dev-app-cli

toc::[]

> This tutorial walks you through deploying services to stand up an application called `national-parks-app` on {product-title} that displays a map of national parks across the world. You will use the {oc-first} to complete this tutorial.

Author: Yeah, I normally would try to avoid the self-referential text in the intros, but I do think it's appropriate here because this is a tutorial. ISG confirms that's it's okay for tutorial content:

> You can use self-referential text in learning modules and tutorials.

So I think I'll leave these as is, but thanks for questioning!

> This tutorial walks you through deploying services to stand up an application called `national-parks-app` on {product-title} that displays a map of national parks across the world. You will use the {oc-first} to complete this tutorial.

Reviewer 2: Good to know that it's allowed in tutorials!

To complete this tutorial, you will perform the following steps:

. xref:../tutorials/dev-app-cli.adoc#getting-started-cli-creating-new-project_dev-app-cli[Create a new project for the application].
+

> This step allows your application to be isolated from other cluster user's workloads.

Reviewer 2: ```suggestion
Create a project so that your application can be isolated from other cluster user's workloads.
```
Would it be incorrect/weird/wrong to say "...can be isolated from other people's workloads."?

. xref:../tutorials/dev-app-cli.adoc#getting-started-cli-granting-permissions_dev-app-cli[Grant view permissions].
+

> This step grants `view` permissions to interact with the OpenShift API to help discover services and other resources running within the project.

Reviewer 2: ```suggestion
Grant `view` permissions so that your project can interact with the OpenShift API to discover services and other resources.
```
Feel free to ignore this suggestion if it's inaccurate. Again, I'm trying to avoid self-referential wording here.

. xref:../tutorials/dev-app-cli.adoc#getting-started-cli-deploying-first-image_dev-app-cli[Deploy the front-end application].
+

> This step deploys the `parksmap` front-end application, exposes it externally, and scales it up to two instances.

Reviewer 2: ```suggestion
Deploy the `parksmap` front-end application, expose it externally, and scale it up to two instances.
```

. xref:../tutorials/dev-app-cli.adoc#getting-started-cli-deploying-python-app_dev-app-cli[Deploy the back-end application].
+

> This step deploys the `nationalparks` back-end application and exposes it externally.

Reviewer 2: ```suggestion
Deploy the `nationalparks` back-end application and expose it externally.
```

. xref:../tutorials/dev-app-cli.adoc#getting-started-cli-connecting-database_dev-app-cli[Deploy the database application].
+

> This step deploys the `mongodb-nationalparks` MongoDB database, loads data into the database, and sets up the necessary credentials to access the database.

Reviewer 2: ```suggestion
Deploy the `mongodb-nationalparks` MongoDB database, load data into the database, and set up the necessary credentials to access the database.
```

. xref:../tutorials/dev-app-cli.adoc#getting-started-cli-view_dev-app-cli[Verify that the map application is available].
+

> This step verifies that you can view the map of national parks in a browser.

Reviewer 2: ```suggestion
Verify that you can view the map of national parks in a browser.
```

[id="before-you-begin_{context}"]
== Before you begin

> Before you start this tutorial, ensure that you have the the following required prerequisites:

Reviewer 2: ```suggestion
Ensure that you have the the following required prerequisites:
```
I think this can be simplified since the name of the section is "Before you begin".

Author: I renamed the heading to "Prerequisites". That's more in line with what our mod docs templates say anyways.

include::modules/getting-started-cli-creating-route-backend.adoc[leveloffset=+2]

[role="_additional-resources"]
.Additional resources
* xref:../cli_reference/openshift_cli/developer-cli-commands.adoc#oc-label[oc label]

// Deploying the database application
include::modules/getting-started-cli-connecting-a-database.adoc[leveloffset=+1]

// Providing access to the database by creating a secret
include::modules/getting-started-cli-creating-secret.adoc[leveloffset=+2]

[role="_additional-resources"]
.Additional resources
* xref:../cli_reference/openshift_cli/developer-cli-commands.adoc#oc-create-secret-generic[oc create secret generic]
* xref:../cli_reference/openshift_cli/developer-cli-commands.adoc#oc-set-env[oc set env]
* xref:../cli_reference/openshift_cli/developer-cli-commands.adoc#oc-rollout-status[oc rollout status]

// Loading data into the database
include::modules/getting-started-cli-load-data-output.adoc[leveloffset=+2]

[role="_additional-resources"]
.Additional resources
* xref:../cli_reference/openshift_cli/developer-cli-commands.adoc#oc-exec[oc exec]

// Displaying the national parks map
include::modules/getting-started-cli-view.adoc[leveloffset=+1]

> // TODO: consider whether to pull down the optional sections (viewing pod, scaling) into a separate bucket below

Author: Great thanks! I'll leave it as is now for this PR, then we can come back and take another look. It might also inspire us too when we start looking into updating the web console tutorial.
